### PR TITLE
Fix "unknown enum constant" warnings

### PR DESCRIPTION
**Pull request by jtraglia**
## PR Description

This gets rid of the "unknown enum constant" warning messages in the build:

```
> Task :eth-benchmark-tests:compileJmhJava
warning: unknown enum constant Status.STABLE
  reason: class file for org.apiguardian.api.API$Status not found
warning: unknown enum constant Status.STABLE
warning: unknown enum constant Status.STABLE
  reason: class file for org.apiguardian.api.API$Status not found
warning: unknown enum constant Status.STABLE
warning: unknown enum constant Status.STABLE
  reason: class file for org.apiguardian.api.API$Status not found
warning: unknown enum constant Status.STABLE
warning: unknown enum constant Status.STABLE
  reason: class file for org.apiguardian.api.API$Status not found
warning: unknown enum constant Status.STABLE
```

We needed to add `apiguardian-api` as a `runtimeOnly` dependency.

See: https://stackoverflow.com/a/46950496

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -730,6 +730,7 @@ subprojects {
 
     runtimeOnly 'org.apache.logging.log4j:log4j-core'
     runtimeOnly 'org.apache.logging.log4j:log4j-slf4j-impl'
+    runtimeOnly 'org.apiguardian:apiguardian-api'
```
Comment (ajsutton) [inline build.gradle]: I suspect we can add this as a testRuntimeOnly if it's required by junit.  Adding it to `runtimeOnly` will cause it to be shipped with Teku which I don't think we want.

**Comment by jtraglia** [comment]
Good call. It still prints the warnings if it is `testRuntimeOnly` but `testFixturesRuntimeOnly` does the trick. Is this fine?

**Comment by ajsutton** [comment]
Yep `testFixtureRuntimeOnly` works fine. Thanks.

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**